Gom4ziz/Source/Domain/RequestArtworkUsecase.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  RequestArtworkUsecase.swift
+//  Gom4ziz
+//
+//  Created by sanghyo on 2022/11/21.
+//
+
+import Foundation
+
+import RxSwift
+
+final class RequestArtworkUsecase {
+    private let artworkRepository: ArtworkRepository
+    
+    init(_ artworkRepository: ArtworkRepository = FirebaseArtworkRepository.shared) {
+        self.artworkRepository = artworkRepository
+    }
+    
+    func requestNextArtwork(_ artworkId: Int) -> Observable<Artwork> {
+        return artworkRepository.requestArtwork(of: artworkId)
```

Comment:
[사소한 의견]
한 줄이라서 return 이 필요 없을 것 같아요 (컴파일러가 반환값 추론가능)

Reply:
[수정 필수]
또한 이 함수 자체가, 다음 질문을 불러와야하는데, 지금 로직은 이 **유즈케이스를 사용하는 쪽에서 다음 질문의 id가 뭔지 계산**해야 하는 것으로 보이네요. 다음 질문의 id를 구하거나, 유저가 아직 그 질문까지는 못 받는다! 이런 로직은 전부 유즈케이스 (도메인) 쪽에서 구현해야 합니다.

따라서 이 함수에선 다음 질문을 불러오기 위한 정보 (User라던가)를 받아서, 자체적으로 다음 질문을 판단한 후 요청해야 할 것으로 보이네요.

```suggestion
func requestNextArtwork(of user: User) -> Observable<Artwork>
```

Reply:
10166bf4b097b13f6f7280d080d96abd500fd984 반영 완료했습니다
미흡한 부분 또 알려주세요 ~ 
empty라는 값은 raywenderlich에서 공부할 때 비슷하게 했던 경험이 있어 사용해봤는데 더 좋은 방법이 있다면 제안 부탁드립니다

Reply:
이것도 error의 종류로 보고 error 처리해줘도 됨! 이라는 가르침